errors.go

```diff
@@ -283,6 +284,20 @@ func (errorType ErrorType) Set(err error) error {
 	}
 }
 
+// Set the type of the error
+func (errorType ErrorType) SetUserMessage(err error, msg string) error {
+	if err == nil {
+		return nil
+	}
+
+	return &customError{
+		error:       errors.WithStack(err),
+		errorType:   errorType,
```

Comment:
Maybe we can inherit the `ErrorType` of `err` if this is otherwise called as `NoType`:
```go
return &customError{
  ...
  errorType: errorType == NoType ? GetType(err) : errorType,
  ...
}
```
Worst case scenario, we end up with a `NoType` error, but if `err` contains a more useful one, then we should probably use that...

Reply:
Good idea. Fixed.

Reply:
I couldn't make it a single line though :(